### DISCUSSION TITLE
How to integrate two pore network models of different sizes？

### DISCUSSION THREAD

**Discussion by Xknightfyy** [PMEAL/OpenPNM#2712] [Q&A] [answered]
Dear professor Jeff Gostick:
Hello! First of all, thank you very much for such a good learning resource. But I've recently encountered some problems with OpenPNM, first of all, how can I generate two pore network models at different scales and merge them together and label them accordingly? I used two pore network models generated connect_pores this built-in function connection, but I can't integrate, can you tell me how? Thank you

**Comment by jgostick** [comment, marked as answer]
We have a function in the topotools module called stitch_networks that should do what your looking for.

**Reply by Xknightfyy** [reply]
This is good information and thank you very much for taking the time to explain. But I still have two small problems, one is how do I change the coordinates of the donated pore network model, so that the two pore network models merge up and down? Second, how do I label two different pore network model areas up, down, left and right?

**Reply by Xknightfyy** [reply]
I have another question about the boundary condition rate, what are the units? I can't use rate very well to describe boundary conditions when electrochemical reactions occur at the boundary.

**Reply by jgostick** [reply]
1. pn1['pore.coords'] += [0, 0, 10], or what ever shift you want
2. pn1['pore.left'] = True, pn2['pore.right'] = True, then when you stitch, these labels will both exist on the big network, but only be True in the locations which were True prior to stitching
3. Units are up to you, but I always do everything in SI.  When you do a diffusion simulation it is mol/s, when you do a flow simulation it is m3/s

**Reply by Xknightfyy** [reply]
Thank you! Thank you for your kind answer! Best wishes to you.

**Reply by Xknightfyy** [reply]
I still have questions about your answer to my second question, I can't just do it for the upper layer set label of the pore network after merging the two pore network models. I output pn1.pore.left and I output the left side of the entire merged pore network model. What should I do to label only the top half? I am  sorry for bothering you with these things, and we hope to hear from you soon.

**Reply by Xknightfyy** [reply]
I think this means that the two remain relatively independent labels after the merger, rather than the overall label after the merger.

**Comment by Xknightfyy** [comment]
<img width="385" alt="ef7eeebdca6ba530424d99ced4a9a58" src="https://user-images.githubusercontent.com/120148734/227110848-ae88396a-7d31-44f3-a7a3-c58f85297690.png">
I try to use the following code to lable the pn1.left, but i failed.

**Reply by likeduck** [reply]
I neet the same question, but when you run a simulation, it was an integral network.
